src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx

```diff
@@ -124,6 +124,8 @@ export default function LineNumberTooltip({
         if (lineNumber === lastHoveredLineNumber.current) {
           if (codeHeatMaps) {
             dispatch(
+              // TODO This is bad API design. Instead of passing in an `onFailure` callback,
+              // this action  should return a promise
```

Comment:
yeah, I wrote this code, there was a reason that using a promise here was really difficult, but I don't remember what it was off the top of my head.